### PR TITLE
[Bug] [lang] Fix augmented assign for sar

### PR DESCRIPTION
**Pull request by strongoier**
### Brief Summary

While I was removing redundant codegen for `shr` (#6134), I found that augmented assign for `sar` was accidentally calling `shr`, so I fixed it by the way.

### CODE REVIEW

File: taichi/transforms/demote_operations.cpp [hunk 1]
```diff
@@ -118,8 +118,8 @@ class DemoteOperations : public BasicStmtVisitor {
                is_signed(lhs->element_type())) {
       // @ti.func
       // def bit_shr(a, b):
-      //     signed_a = ti.cast(a, ti.uXX)
-      //     shifted = ti.bit_sar(signed_a, b)
+      //     unsigned_a = ti.cast(a, ti.uXX)
+      //     shifted = ti.bit_sar(unsigned_a, b)
```
Comment (lin-hitonami) [inline taichi/transforms/demote_operations.cpp:122]: Why is it sar here?

File: tests/python/test_bit_operations.py [hunk 2]
```diff
@@ -13,8 +13,14 @@ def test_bit_shl():
...
 
+    @ti.kernel
+    def shl_assign(a: ti.i32, b: ti.i32) -> ti.i32:
+        c = a
+        c <<= b
+        return c
+
     for i in range(8):
-        assert shl(3, i) == 3 * 2**i
+        assert shl(3, i) == shl_assign(3, i) == 3 * 2**i
```
Comment (jim19930609) [inline tests/python/test_bit_operations.py:23]: `3 <<= 1` gives the same result regardless of whether it's "signed shift" or "unsigned shift". Shall we test with negative number or positive number large enough(`01xxxxxx`) to influence the sign-bit after shift?

File: taichi/transforms/demote_operations.cpp (hunk 1 above)
Comment (strongoier) [reply taichi/transforms/demote_operations.cpp:122]: `sar` for unsigned is equivalent to `shr`.

File: tests/python/test_bit_operations.py (hunk 2 above)
Comment (strongoier) [reply tests/python/test_bit_operations.py:23]: `left shift` gives the same result anyway. Only `right shift` will be influenced by the sign bit.

Comment (lin-hitonami) [reply tests/python/test_bit_operations.py:23]: IIRC shl and sal are the same, and they do not preserve the signed bit.

File: taichi/transforms/demote_operations.cpp (hunk 1 above)
Comment (strongoier) [reply taichi/transforms/demote_operations.cpp:122]: I guess the main goal of the demotion is to make codegen simpler (only `sar` needed).

**Review by lin-hitonami — APPROVED** [review]
LGTM

**Pull request merged**